externs/firebase-app-externs.js

```diff
@@ -275,3 +275,26 @@ firebase.Promise.reject = function(error) {};
  * @return {!firebase.Promise<!Array<*>>}
  */
 firebase.Promise.all = function(values) {};
+
+/** @interface */
+firebase.Observer = function() {};
+
+/**
+ * @template V
+ * @param {?V} value
```

Comment:
Just one last change, put the "@template V" and "@template E" annotations in the same comment block as the "@interface" instead of on the individual methods.